File: functions/alsa-lib/variants/go/4_val_seg_2.go
package main

func valSeg(val int32) int {
	r := 1
	val >>= 8
	switch {
	case val & 0xf0 != 0:
		val >>= 4
		r += 4
		fallthrough
	case val & 0x0c != 0:
		val >>= 2
		r += 2
		fallthrough
	case val & 0x02 != 0:
		r++
	}
	return r
}


func main(){
    valSeg(0)
}

File: functions/alsa-lib/variants/c/4_val_seg_2.c
#include <stdint.h>

static inline int val_seg(int val)
{
	int r = 1;
	if ((val>>12) & 0x0f) {
		r += 6;
	} else if ((val>>8) & 0x0f) {
		r += 4;
		if ((val>>4) & 0x03) {
			r += 2;
		}
	}
	if ((val>>8) & 0x02) {
		r += 1;
	}
	return r;
}


int main(){
    val_seg(0);
}